### PR TITLE
add ai-hint in hints view

### PR DESCRIPTION
**Pull request by smallst**
fix ENG-706

add ai-hint button and also adjust hints height
image
image

## Summary by CodeRabbit

- **New Features**
  - Introduced AI bot help functionality in the hints section, providing users with additional assistance if needed.

- **Style**
  - Updated styles for the hints view to improve layout and usability.

- **Bug Fixes**
  - Adjusted hint body height to ensure better display across different screen sizes.

- **Chores**
  - Added new string key for "Want more hints?" to support AI bot help feature.

### CODE REVIEW

File: app/styles/play/level/hints-view.sass [hunk 1]
```diff
@@ -54,3 +55,21 @@
...
+    border-top: 1px solid white
+    padding-top: 10px
+    margin-top: 10px
+
+    img.avatar
+      width: 40px
+      margin-right: 12px
+
+    button
+      margin-left: 6px
```
Comment (mrfinch) [inline app/styles/play/level/hints-view.sass:70]: we dont have an convention of sizes but I generally prefer margin and padding to be multiple of 5s and font-sizes to be multiple of 2

File: app/views/play/level/HintsView.coffee [hunk 2]
```diff
@@ -33,6 +35,16 @@ module.exports = class HintsView extends CocoView
     @listenTo(@hintsState, 'change', debouncedRender)
     @listenTo(@state, 'change:hintIndex', @updateHint)
     @listenTo(@hintsState, 'change:hidden', @visibilityChanged)
+    @showAiBotHelp = false
+    console.log("???Kjla,", @aceConfig.levelChat)
```
Comment (mrfinch) [inline app/views/play/level/HintsView.coffee:39]: remove

File: app/views/play/level/HintsView.coffee [hunk 3]
```diff
@@ -33,6 +35,16 @@ module.exports = class HintsView extends CocoView
...
+    if @aceConfig.levelChat != 'none'
+      if me.isAdmin()
+        @showAiBotHelp = true
+      else if me.isHomeUser() && me.getLevelChatExperimentValue() == 'beta'
+        @showAiBotHelp = true
+      else if not me.isHomeUser()
+        @showAiBotHelp = true
+    console.log("???Kjla,", @showAiBotHelp)
```
Comment (mrfinch) [inline app/views/play/level/HintsView.coffee:47]: remove

File: app/views/play/level/HintsView.coffee [hunk 4]
```diff
@@ -33,6 +35,16 @@ module.exports = class HintsView extends CocoView
     @listenTo(@hintsState, 'change', debouncedRender)
     @listenTo(@state, 'change:hintIndex', @updateHint)
     @listenTo(@hintsState, 'change:hidden', @visibilityChanged)
+    @showAiBotHelp = false
+    console.log("???Kjla,", @aceConfig.levelChat)
+    if @aceConfig.levelChat != 'none'
```
Comment (mrfinch) [inline app/views/play/level/HintsView.coffee:40]: i think we have this logic in some other file too, can we refactor it to lib/util to like `shouldShowAiBotHelp`

File: app/views/play/level/HintsView.coffee [hunk 5]
```diff
@@ -106,3 +118,9 @@ module.exports = class HintsView extends CocoView
       @state.set('hintsUsed', hintsUsed)
       clearInterval(@timerIntervalID)
     @state.set('hintsViewTime', hintsViewTime)
+
+  onAIHelpClicked: (e) ->
+    rand = _.random(1, 13)
+    message = $.i18n.t('ai.prompt_level_chat_' + rand)
+    Backbone.Mediator.publish 'level:add-user-chat', { message }
+    _.delay (=> @handleUserCreditsMessage()), 5000
```
Comment (mrfinch) [inline app/views/play/level/HintsView.coffee:126]: where is this function defined?

Comment (smallst) [reply app/views/play/level/HintsView.coffee:126]: oh good question.

File: app/views/play/level/HintsView.coffee (hunk 4 above)
Comment (smallst) [reply app/views/play/level/HintsView.coffee:40]: ya sure

File: app/core/utils.js [hunk 6]
```diff
@@ -2373,6 +2373,21 @@ const getUserTimeZone = function (user) {
   }
 }
 
+const shouldShowAiBotHelp = function (aceConfig) {
+  if (aceConfig.levelChat != 'none') {
+    if(me.isAdmin()) {
+      return true
+    }
+    else if(me.isHomeUser() && me.getLevelChatExperimentValue() == 'beta') {
```
Comment (mrfinch) [inline app/core/utils.js:2381]: why is else if on next line? 🤔

**Review by mrfinch — APPROVED** [review]

File: app/core/utils.js (hunk 6 above)
Comment (smallst) [reply app/core/utils.js:2381]: lol interesting. i think eslint would fix it but seems not?

**Pull request merged**